### PR TITLE
Dependency injection

### PR DESCRIPTION
**Pull request by crmsnwl**

### CODE REVIEW

File: CityPuzzle/Classes/IUserVerifier.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿namespace CityPuzzle.Classes
+{
+    public interface IUserVerifier
+    {
+        bool CPass(string name, string pass);
```
Comment (AllBill127) [inline CityPuzzle/Classes/IUserVerifier.cs:5]: Add comments to specify what these methods do as theyr names ar not very descriptive

File: CityPuzzle/Classes/UserVerifier.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using BCryptNet = BCrypt.Net.BCrypt;
+
+namespace CityPuzzle.Classes
+{
+    public class UserVerifier : IUserVerifier
+    {
+        public Boolean CPass(string name, string pass)
```
Comment (AllBill127) [inline CityPuzzle/Classes/UserVerifier.cs:11]: Bad data type: Boolean - should be lowercase

File: CityPuzzle/Classes/UserVerifier.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            if (String.IsNullOrWhiteSpace(name) || String.IsNullOrWhiteSpace(pass))
+            {
+                return false;
+            }
+            // pass = PassToHash(pass);//     //                              //encrypt
```
Comment (AllBill127) [inline CityPuzzle/Classes/UserVerifier.cs:17]: Delete unnecessary comment

File: CityPuzzle/Classes/UserVerifier.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            var info = Sql.ReadUsers();
+            App.CurrentUser = info.SingleOrDefault(x => x.UserName.ToLower().Equals(name.ToLower()) && PassVer(pass, x.Pass));
+            return App.CurrentUser != null;
+        }
+        public Boolean CUser(string name)
```
Comment (AllBill127) [inline CityPuzzle/Classes/UserVerifier.cs:22]: Fix data type

**Comment by crmsnwl** [comment]
Fixed commented problems and resolved merge conflicts.

File: CityPuzzle/Classes/IUserVerifier.cs [hunk 5]
```diff
@@ -2,9 +2,13 @@
 {
     public interface IUserVerifier
     {
+        //Called by CheckPassword in User.cs
```
Comment (AllBill127) [inline CityPuzzle/Classes/IUserVerifier.cs:5]: Reverse information. Write what interface methods do

**Comment by AllBill127** [comment]
After fixing interface comments merge and delete branch

**Comment by AllBill127** [comment]
If no more changes will be done to dependency injection, you can merge and delete this branch

**Review by AllBill127 — APPROVED** [review]
Good comments

**Pull request merged**